todo_app/app.py

```diff
@@ -25,9 +48,20 @@ def index():
             items.sort(key=lambda x: x.due_date)
         elif request.values.get('sort') == '2':
             items.sort(key=lambda x: x.due_date, reverse=True)
-        return render_template("index.html",View_Model=get_view_model, todays_date=todays_date)
+
+        if hasattr('current_user', 'role'):
+            if current_user.role == 'writer':
+                return render_template("index.html",View_Model=get_view_model, todays_date=todays_date)
+            elif current_user.role == 'reader':
+                return render_template('index_read.html', View_Model=get_view_model, todays_date=todays_date)
+            else:
+                return render_template('index.html', View_Model=get_view_model, todays_date=todays_date)
+        else:
+            return render_template('index.html', View_Model=get_view_model, todays_date=todays_date)
```

Comment:
I suspect this is meant to deal with the case for testing? If so I'd be inclined to program this a bit more defensively to make that obvious and rather than just letting it fall back to this if the user object has its role removed; e.g. checking if the LOGIN_DISABLED config is set and then returning the write template, otherwise checking the role?